cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KRaftMetadataManager.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.api.kafka.model.status.KafkaStatus;
+import io.strimzi.operator.cluster.model.KafkaCluster;
+import io.strimzi.operator.common.AdminClientProvider;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.VertxUtil;
+import io.strimzi.operator.common.model.StatusUtils;
+import io.strimzi.operator.common.operator.resource.SecretOperator;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.FeatureUpdate;
+import org.apache.kafka.clients.admin.UpdateFeaturesOptions;
+import org.apache.kafka.server.common.MetadataVersion;
+
+import java.util.Map;
+
+/**
+ * Utility class for managing KRaft the metadata version. It encapsulates the methods for getting the current metadata
+ * version and updating it.
+ */
+public class KRaftMetadataManager {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(KRaftMetadataManager.class.getName());
+    /* test */ static final String METADATA_VERSION_KEY = MetadataVersion.FEATURE_NAME;
+
+    /**
+     * Method for managing the KRaft metadata version. This method will get the current metadata version and if needed,
+     * upgrade or downgrade it. It will also update the currently used metadata version in the Kafka CR status. This
+     * method might fail when it fails to communicate with the Kafka cluster and get the metadata version. But it will
+     * not fail when the update of the metadata version fails in order to not fail the whole reconciliation. Such event
```

Comment:
I think I have got what you are explaining but the usage of verb "fail" so many times makes it harder to be readable.

Reply:
I tried to reword it. Please let me know if it feels a bit more clear.

Reply:
Yeah, it looks great now thanks!

Reply:
@ppatierno So, can you approve the PR? 😉